drivers/sensor/ist8310/ist8310.c

```diff
@@ -0,0 +1,238 @@
+/* ist8310.c - Driver for Isentek IST8310 Geomagnetic Sensor */
+
+/*
+ * Copyright (c) 2023 NXP Semiconductors
+ * Copyright (c) 2023 Cognipilot Foundation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+#include "ist8310.h"
+
+LOG_MODULE_REGISTER(IST8310, CONFIG_SENSOR_LOG_LEVEL);
+
+static inline int ist8310_bus_check(const struct device *dev)
+{
+	const struct ist8310_config *cfg = dev->config;
+
+	return cfg->bus_io->check(&cfg->bus);
+}
+
+static inline int ist8310_reg_read(const struct device *dev,
+				  uint8_t start, uint8_t *buf, int size)
+{
+	const struct ist8310_config *cfg = dev->config;
+
+	return cfg->bus_io->read(&cfg->bus, start, buf, size);
+}
+
+static inline int ist8310_reg_write(const struct device *dev, uint8_t reg,
+				   uint8_t val)
+{
+	const struct ist8310_config *cfg = dev->config;
+
+	return cfg->bus_io->write(&cfg->bus, reg, val);
+}
+
+static int ist8310_sample_fetch(const struct device *dev,
+			       enum sensor_channel chan)
+{
+
+	struct ist8310_data *drv_data = dev->data;
+	uint8_t buff[6];
+
+	if (ist8310_reg_read(dev, IST8310_STATUS_REGISTER1, (uint8_t *)buff, 1) < 0) {
+		LOG_ERR("failed to read status register 1");
+		return -EIO;
+	}
+
+	if ((buff[0] & STAT1_DRDY) == 0) {
+		LOG_ERR("Data not ready");
+		if (ist8310_reg_write(dev, IST8310_CONTROL_REGISTER1, CTRL1_MODE_SINGLE) < 0) {
+			LOG_ERR("failed to set single");
+			return -EIO;
+		}
+		return -EIO;
+	}
+
+	if (ist8310_reg_read(dev, IST8310_OUTPUT_VALUE_X_L, (uint8_t *)buff, 6) < 0) {
+		LOG_ERR("failed to read mag values");
+		return -EIO;
+	}
+
+	drv_data->sample_x  = (buff[1] << 8 | buff[0]);
+	drv_data->sample_y  = (buff[3] << 8 | buff[2]);
+	drv_data->sample_z  = (buff[5] << 8 | buff[4]);
```

Comment:
I think these are missing a call to `sys_be16_to_cpu`. Should look like:

```c
drv_data->sample_x = sys_be16_to_cpu((buff[1] << 8) | buff[0]);
```

Reply:
Good point missed that one, made it more portable now.